### PR TITLE
Update [MobileStepper] component (docs & examples)

### PR DESCRIPTION
**Pull request by manojleaton**
Fixes #BLUI-4373 .

#### Changes proposed in this Pull Request:
- Updated MobileStepper component with MD3Theme
- Please verify this PR for kitchen sink component updates [KitchenSink](https://github.com/etn-ccis/blui-react-native-showcase-demo/pull/89)

#### Screenshots / Screen Recording (if applicable)
Screenshot 2023-10-10 at 1 12 51 PM

#### To Test:
- yarn start

#### Any specific feedback you are looking for?
-

### CODE REVIEW

**Comment by huayunh** [comment]
please also post screenshots here

**Comment by manojleaton** [comment]
> please also post screenshots here

Yes will add screenshots and some more details, make it ready for review

File: components/src/md3/MobileStepper/MobileStepper.tsx [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+                                    active ? styles.filled : {},
+                                ]}
+                            />
+                        );
+                    })}
+
+                {variant === 'text' && (
+                    <Body1 style={[defaultStyles.text, styles.text]}>
+                        {adjustedActiveStep + 1} / {adjustedSteps}
+                    </Body1>
```
Comment (ArshdeepSingh3-eaton) [inline components/src/md3/MobileStepper/MobileStepper.tsx:177]: Remove Body1 component and use Text Component instead

**Review by ArshdeepSingh3-eaton — CHANGES_REQUESTED** [review]
Remove Body1 component and use text Component

File: components/src/md3/MobileStepper/MobileStepper.tsx [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+ *
+ * This component is used to show progress through a set of pages. It displays
+ * the total number of pages or steps and which one is currently being displayed.
+ *
+ * This is based on the [MobileStepper](https://material-ui.com/components/steppers/#mobile-stepper) component from Material UI.
+ */
+export const MobileStepper: React.FC<MobileStepperProps> = (props) => {
+    const theme = useTheme(props.theme);
+    const {
+        activeColor = getPrimary500(theme) || theme.colors.primary,
```
Comment (ArshdeepSingh3-eaton) [inline components/src/md3/MobileStepper/MobileStepper.tsx:136]: set `activeColor = theme.colors.primary,`

> Remove the getPrimary500(theme)

File: components/src/md3/MobileStepper/MobileStepper.tsx [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+        circle: {
+            height: 8,
+            width: 8,
+            borderRadius: 8,
+            marginHorizontal: 4,
+            overflow: 'hidden',
+            backgroundColor: props.inactiveColor || (theme.dark ? theme.colors.onBackground : Colors.gray[200]),
+        },
+        filled: {
+            backgroundColor: props.activeColor || (theme.dark ? theme.colors.primary : theme.colors.primary),
```
Comment (huayunh) [inline components/src/md3/MobileStepper/MobileStepper.tsx:42]: This tertiary operator is not doing anything

File: components/src/md3/MobileStepper/MobileStepper.tsx [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+            alignItems: 'center',
+            justifyContent: 'center',
+        },
+        circle: {
+            height: 8,
+            width: 8,
+            borderRadius: 8,
+            marginHorizontal: 4,
+            overflow: 'hidden',
+            backgroundColor: props.inactiveColor || (theme.dark ? theme.colors.onBackground : Colors.gray[200]),
```
Comment (huayunh) [inline components/src/md3/MobileStepper/MobileStepper.tsx:39]: Use the custom "disabled" color for the background color

File: components/src/md3/MobileStepper/MobileStepper.tsx (hunk 3 above)
Comment (manojleaton) [reply components/src/md3/MobileStepper/MobileStepper.tsx:42]: Removed tertiary operator from here, Thanks

File: components/src/md3/MobileStepper/MobileStepper.tsx (hunk 4 above)
Comment (manojleaton) [reply components/src/md3/MobileStepper/MobileStepper.tsx:39]: There is no colour named "disabled", but onBackground colour is same colour as the disabled colour.

Comment (huayunh) [reply components/src/md3/MobileStepper/MobileStepper.tsx:37]: (1)
> There is no colour named "disabled"

Yes, that is a custom-added color from the design side. If that variable does not exist in the theme, see if you can find a way to add that to the theme, or at least leave a comment here to update the color in the future.

(2) 
> onBackground colour is same colour as the disabled colour
This is what "onBackground" currently looks like on the design side. 

image

**Comment by JeffGreiner-eaton** [comment]
https://github.com/etn-ccis/blui-react-native-showcase-demo/pull/89#issuecomment-1758451100

File: components/src/md3/MobileStepper/MobileStepper.tsx (hunk 4 above)
Comment (manojleaton) [reply components/src/md3/MobileStepper/MobileStepper.tsx]: Added custom disabled color variable with the Hex value mentioned above in MD3BluiLightTheme file. It's picking up the added color, but showing error as below so for that added a comment in the code.
Screenshot 2023-10-12 at 4 24 47 PM

**Review by ArshdeepSingh3-eaton — APPROVED** [review]

**Review by JeffGreiner-eaton — APPROVED** [review]
looks good to me, how are we tracking the additions needed in RN Theme for this theme.colors.disabled?

**Pull request merged**